lib/models/Rock.js

```diff
@@ -0,0 +1,70 @@
+const pool = require('../utils/pool');
+
+module.exports = class Rock {
+  id;
+  name;
+  crystal_system;
+  hardness;
+
+  constructor(row) {
+    this.id = row.id;
+    this.name = row.name;
+    this.crystal_system = row.crystal_system;
+    this.hardness = row.hardness;
+  }
+  static async findAll() {
+    const { rows } = await pool.query('SELECT * FROM rocks');
+    return rows.map((row) => new Rock(row));
+  }
+
+  static async insert({name, crystal_system, hardness}) {
+    const { rows } = await pool.query(
+      `INSERT INTO 
+        rocks (name, crystal_system, hardness)
+        VALUES
+          ($1, $2, $3)
+        RETURNING
+        *
+        `,
+        [name, crystal_system, hardness]
+        );
```

Comment:
We typically put early returns here if nothing was returned from our database query to prevent errors. something like `if(!rows[0]) return null`